modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -598,6 +634,7 @@ public void start() {
     public void beforeNodeStop() {
         orphanDetector.stop();
         txStateVolatileStorage.stop();
+        rwTxCounterByCatalogVersion.clear();
```

Comment:
Why do we need this? Will not this allow a false 'yes, all transactions have finished' response to be sent, even though some transactions are still pending?

Reply:
I think you're right, I'll move it to TxManagerImpl#stop.